### PR TITLE
blackbox-terminal: init at 0.12.0

### PR DESCRIPTION
**Pull request by chuangzhu**
###### Description of changes

Solves #188062. Depends on #182618. ~~The packaging work is not perfect currently. It doesn't always launch. Most of the times it just crashes with the following error message:~~

<del>

```
**
GLib:ERROR:../glib/gvarianttypeinfo.c:186:g_variant_type_info_check: assertion failed: (0 <= index && index < 24)
Bail out! GLib:ERROR:../glib/gvarianttypeinfo.c:186:g_variant_type_info_check: assertion failed: (0 <= index && index < 24)
Aborted (core dumped)
```
</del>

2022-09-02T17:41:17,572036917+08:00

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by wyndon** [comment]
This is the issue you're facing : https://github.com/mitchmindtree/blackbox.nix/issues/2

Patch is provided in this PR : https://github.com/mitchmindtree/blackbox.nix/pull/3

**Comment by SamLukeYes** [comment]
vte 0.70 and libadwaita 1.2 can be found in https://github.com/NixOS/nixpkgs/pull/182618

**Comment by chuangzhu** [comment]
Rebased to #182618. No patch and overrideAttrs is needed anymore. The app works perfectly.

File: pkgs/applications/terminal-emulators/blackbox-terminal/marble.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    rev = "6dcc6fefa35f0151b0549c01bd774750fe6bdef8";
+    sha256 = "sha256-0VJ9nyjWOOdLBm3ufleS/xcAS5YsSedJ2NtBjyM3uaY=";
+  };
+
+  nativeBuildInputs = [
+    meson
+    ninja
+    pkg-config
+    vala
+    gtk3
```
[the remark this answers is not in the document]
Comment (chuangzhu) [reply pkgs/applications/terminal-emulators/blackbox-terminal/marble.nix:32]: On the `wip/gtk4` branch, gtk3 is only required for the `gtk-update-icon-cache` command. If gtk3 is omitted, a error is thrown at install phase, not build phase.

```
> FileNotFoundError: [Errno 2] No such file or directory: 'gtk-update-icon-cache'
>
> FAILED: install script '/nix/store/sz0j8k8ljh7y8qgyfxgqb3ws11bcy4gs-python3-3.10.6/bin/python3.10 /build/source/build-aux/meson/postinstall.py 42.alpha0' exit code 1, stopped
```

[the remark this answers is not in the document]
Comment (jtojnar) [reply pkgs/applications/terminal-emulators/blackbox-terminal/marble.nix]: This should be fixed upstream by switching to `meson.post_install()` which supports both gtk3 and gtk4. Example: https://gitlab.gnome.org/GNOME/gnome-logs/-/merge_requests/46

**Comment by schuelermine** [comment]
Thank you very much for this PR

**Pull request merged**